compiler+runtime/src/cpp/jank/runtime/obj/persistent_string.cpp

```diff
@@ -62,6 +63,41 @@ namespace jank::runtime::obj
     return data.compare(s.data);
   }
 
+  object_ptr persistent_string::get(object_ptr const key) const
+  {
+    return get(key, nil::nil_const());
+  }
+
+  object_ptr persistent_string::get(object_ptr const key, object_ptr const fallback) const
+  {
+    if(key->type == object_type::integer)
+    {
+      auto const i(static_cast<size_t>(expect_object<integer>(key)->data));
+      if(data.size() <= i)
+      {
+        return nil::nil_const();
+      }
+      return make_box<runtime::obj::character>(data[i]);
+    }
+    return fallback;
+  }
+
+  native_bool persistent_string::contains(object_ptr const key) const
+  {
+    if(key->type == object_type::integer)
+    {
+      auto const i(static_cast<size_t>(expect_object<integer>(key)->data));
+      return 0 <= i && i < data.size();
```

Comment:
`i is always > 0` since `i` is unsigned here. We'll need the `< 0` check to be done prior to the cast.

Reply:
Ah, I based it on persistent_vector.cpp, does that need the same fix? The vector `contains` impl looks more like what you're saying is correct but both `get` and `get_entry` seem suspicious.

https://github.com/jank-lang/jank/blob/86c16899c446f5ef5d54cf3c6317e7290dd72ca8/compiler%2Bruntime/src/cpp/jank/runtime/obj/persistent_vector.cpp#L232

Reply:
Actually maybe it's just the `get` without fallback on `persistent_vector.cpp` that seems off.

Reply:
Yeah, I'd say so. Better to have more checks to start with.

Reply:
I found the same problem in several other files and attempted fixes.